lib.php

```diff
@@ -663,12 +663,12 @@ function diary_print_overview($courses, &$htmlarray) {
         }
 
         if ($diaryopen) {
-            $url = new moodle_url('mod/diary/view.php', ['id' => $diary->coursemodule]);
+            $url = new moodle_url('/mod/diary/view.php', ['id' => $diary->coursemodule]);
+            $link = html_writer::link($url, $diary->name, ['class' => $diary->visible ? '' : 'dimmed']);
```

Comment:
does `use moodle_url` and `use html_writer` need to be added to the top of this page ?

Reply:
Adding these to the top of the page seems to break the site:
```
Warning: The use statement with non-compound name 'moodle_url' has no effect in /var/www/evidence-central/mod/diary/lib.php on line 26

Warning: The use statement with non-compound name 'html_writer' has no effect in /var/www/evidence-central/mod/diary/lib.php on line 27
```

Reply:
Adding these to the top of the page seems to break the site:
```
Warning: The use statement with non-compound name 'moodle_url' has no effect in /var/www/evidence-central/mod/diary/lib.php on line 26

Warning: The use statement with non-compound name 'html_writer' has no effect in /var/www/evidence-central/mod/diary/lib.php on line 27
```

Reply:
Ah yeah probably because this file isn't namespaced - don't worry about it then